### PR TITLE
allow style attribute in table

### PR DESCRIPTION
**Pull request by OsmondJiang**
https://dev.azure.com/ceapex/Engineering/_workitems/edit/114684?src=WorkItemMention&src-action=artifact_link

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/docfx/pull/5089)

### CODE REVIEW

**Comment by OsmondJiang** [comment]
Why this diff is not found in migration test? Seems v2 always has the style attributes

image


@yufeih  @928PJY

**Comment by OsmondJiang** [comment]
> Why this diff is not found in migration test? Seems v2 always has the style attributes
> 
> image
> 
> @yufeih @928PJY

This need to be fixed in Migration Tool normalizing html

File: docs/specs/markdown.yml [hunk 1]
```diff
@@ -297,6 +297,49 @@ outputs:
   docs/a.json: |
     { "conceptual": "<div></div><p>body</p>" }
 ---
+# table with styles is allowed
+inputs:
+  docfx.yml:
+  docs/a.md: |
```
Comment (yufeih) [inline docs/specs/markdown.yml]: consider put the test in a `HtmlUtilityTest.HtmlStripTags` UT

File: src/docfx/lib/HtmlUtility.cs [hunk 2]
```diff
@@ -15,6 +16,8 @@ internal static class HtmlUtility
         private static readonly Func<HtmlAttribute, int> s_getValueStartIndex =
             ReflectionUtility.CreateInstanceFieldGetter<HtmlAttribute, int>("_valuestartindex");
 
+        private static readonly Regex s_styleCellAlignRegEx = new Regex("text-align: (?:(?:left)|(?:center)|(?:right));", RegexOptions.Compiled);
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs]: handle `style="text-align: left; background-color: yellow"` case

File: docs/specs/markdown.yml (hunk 1 above)
Comment (OsmondJiang) [reply docs/specs/markdown.yml]: it's the case like spec test `do not allow custom js, styles and css`, so keep it in spec test may be ok?

File: src/docfx/lib/HtmlUtility.cs [hunk 3]
```diff
@@ -15,6 +16,8 @@ internal static class HtmlUtility
         private static readonly Func<HtmlAttribute, int> s_getValueStartIndex =
             ReflectionUtility.CreateInstanceFieldGetter<HtmlAttribute, int>("_valuestartindex");
 
+        private static readonly Regex s_styleCellAlignRegEx = new Regex("^text-align: (?:(?:left)|(?:center)|(?:right));$", RegexOptions.Compiled);
```
Comment (yufeih) [inline src/docfx/lib/HtmlUtility.cs]: Looks like this regex means it must be exactly `text-align: left;` or `text-align: right;` or `text-align: center;`, so may be a `Contains` is enough.

On the other hand, `text-align:right` is also a valid style, but I don't think we need to go that far.

File: docs/specs/markdown.yml (hunk 1 above)
Comment (yufeih) [reply docs/specs/markdown.yml]: Yeah, the table is fine, just those other tests

File: test/docfx.Test/lib/HtmlUtilityTest.cs [hunk 4]
```diff
@@ -1,6 +1,7 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using Markdig.Syntax.Inlines;
```
Comment (yufeih) [inline test/docfx.Test/lib/HtmlUtilityTest.cs]: remove

File: src/docfx/lib/HtmlUtility.cs (hunk 3 above)
Comment (OsmondJiang) [reply src/docfx/lib/HtmlUtility.cs]: `Contains` or `Equals`?

Comment (yufeih) [reply src/docfx/lib/HtmlUtility.cs]: I mean `Array.Contains`, you have an array of `['text-align: left', 'text-align: right', etc]`

**Pull request merged**